src/bundle.rs

```diff
@@ -394,7 +465,36 @@ impl<T: Authorization, V> Bundle<T, V> {
     }
 }
 
-impl<T: Authorization, V: Copy + Into<i64>> Bundle<T, V> {
+pub(crate) fn derive_bvk<A, V: Clone + Into<i64>, P: OrchardPrimitives>(
+    actions: &NonEmpty<Action<A, P>>,
+    value_balance: V,
+    burn: &[(AssetBase, NoteValue)],
+) -> redpallas::VerificationKey<Binding> {
+    let cv_nets: Vec<_> = actions.into_iter().map(|a| a.cv_net().clone()).collect();
+    derive_bvk_raw(&cv_nets, ValueSum::from_raw(value_balance.into()), burn)
+}
+
+pub(crate) fn derive_bvk_raw(
+    cv_nets: &[ValueCommitment],
+    value_balance: ValueSum,
+    burn: &[(AssetBase, NoteValue)],
+) -> redpallas::VerificationKey<Binding> {
+    (cv_nets.iter().sum::<ValueCommitment>()
```

Comment:
Please do not delete consensus rule references.
```suggestion
    // https://p.z.cash/TCR:bad-txns-orchard-binding-signature-invalid?partial
    (cv_nets.iter().sum::<ValueCommitment>()
```